zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/usertask/UserTaskProcessor.java

```diff
@@ -127,6 +129,20 @@ private void handleCommandProcessing(
     final var eventType = mapIntentToEventType(intent);
 
     if (userTaskElement.hasTaskListeners(eventType)) {
+      /*
+       * Store the original command's metadata (requestId, requestStreamId) before creating
+       * the first task listener job. This metadata will be used later to finalize the original
+       * command after all task listeners have been processed, ensuring that the engine can respond
+       * appropriately to the original command request.
+       *
+       * Note:
+       * Typically, persistence logic should be handled via `*Applier` classes. However, since
+       * this involves request-related data, so in the case of command reconstruction,
+       * we will have new request values anyway, so persisting these data here is acceptable.
+       * A similar approach has been used in `ProcessInstanceCreationCreateWithResultProcessor`.
+       */
+      storeUserTaskRecordRequestMetadata(command);
```

Comment:
👍 I like that you store it here. This ensures it's stored for all user task commands, rather than just for the complete command.